### PR TITLE
Enable transitive project references

### PR DESCRIPTION
**Pull request by jaredpar**
When moving to the new SDK I initially disabled transitive project
references because it was causing issues building our VSIX projects. Had
some time this week to sit down and work through the problem which comes
down to the VSIX indirectly referencing a multi-targetted project. This
is a known bug in the SDK that can be worked around by doing the
following:

- Explicitly referencing the project
- Using AdditionalProperties to specify the TF to use
- Using Name to work around inability of VSSDK to infer Name

The SDK bug is here.

https://github.com/dotnet/sdk/issues/433

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure, @nguerrera

File: src/VisualStudio/IntegrationTest/TestSetup/VisualStudioIntegrationTestSetup.csproj [hunk 1]
```diff
@@ -74,6 +74,11 @@
     <ProjectReference Include="..\TestUtilities\VisualStudioIntegrationTestUtilities.csproj">
       <Name>VisualStudioIntegrationTestUtilities</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\..\Test\Utilities\Portable\TestUtilities.csproj">
+      <!-- https://github.com/dotnet/sdk/issues/433 -->
+      <AdditionalProperties>TargetFramework=net461</AdditionalProperties>
+      <Name>TestUtilities</Name>
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/TestSetup/VisualStudioIntegrationTestSetup.csproj]: :bulb: consider automated placement of the `Name` property:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/7cf733282f0a76f9235de7ea24cfcd6efbb2cf17/StyleCop.Analyzers/StyleCop.Analyzers.Vsix/StyleCop.Analyzers.Vsix.csproj#L51-L52

Comment (jaredpar) [reply src/VisualStudio/IntegrationTest/TestSetup/VisualStudioIntegrationTestSetup.csproj]: I saw that example (actually how I figured out how to fix this bug). Decided to keep this minimal for now as I have much bigger changes that are riding on top of this fix. Essentially moving the Roslyn repo completely over to Directory.build.props + new MSBuild file format.

**Comment by jaredpar** [comment]
After spending a lot of time trying to get this to pass on Jenkins I've concluded that implicit transitive references simply aren't a good fit for Roslyn. Instead I'm asking the SDK to make opting out of them a supported feature. 

https://github.com/dotnet/sdk/issues/1750

Summary of the problems (more details in linked issue):

- Too much incompatibility with tooling: explodes number of references, requires use of new data, conflicts with multi-targeting. 
- Performance: when enabled on my machine I saw a 31% increase in build time for no specific feature gain.

**Pull request closed without merging**